manifests/plugin/discovery.pp

```diff
@@ -33,15 +33,19 @@
   if $install_images {
     include ::tftp::params
 
-    foreman::remote_file {"${::tftp::params::root}boot/${kernel}":
-      remote_location => "${source}${kernel}",
+    foreman::remote_file {"${::tftp::params::root}boot/${image_name}":
+      remote_location => "${source}${image_name}",
       mode            => 0644,
       require         => File["${::tftp::params::root}boot"],
+      notify          => Exec["untar ${image_name}"],
     }
-    foreman::remote_file {"${::tftp::params::root}boot/${initrd}":
-      remote_location => "${source}${initrd}",
-      mode            => 0644,
-      require         => File["${::tftp::params::root}boot"],
+
+    exec { "untar ${image_name}":
+      command     => "tar xf ${image_name}",
+      path        => '/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin',
+      cwd         => "${::tftp::params::root}boot",
+      refreshonly => true,
```

Comment:
can we use `creates =>` with a filename, or does the filename inside the tarball vary?